source/appModules/calculator.py

```diff
@@ -0,0 +1,88 @@
+#appModules/calculator.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2019 NV Access Limited, Joseph Lee
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""App module for Windows 10 Calculator"""
+
+import appModuleHandler
+import api
+import controlTypes
+from NVDAObjects.UIA import UIA
+import queueHandler
+import ui
+import scriptHandler
+
+# Filter live region change elements to avoid repeated announcements.
+# A dedicated function is provided to react to future Calculator changes.
+def shouldLiveRegionChangeProceed(obj):
+	automationID = obj.UIAElement.cachedAutomationID
+	if automationID == "DateDiffAllUnitsResultLabel":
+		return True
+	elif automationID == "":
+		prevAutomationID = obj.previous.UIAElement.cachedAutomationID
+		if prevAutomationID == "negateButton":
+			return False
+		elif prevAutomationID == "numberPad":
+			return api.getForegroundObject().children[1].children[3].UIAElement.cachedAutomationID != "CalculatorResults"
+	return False
+
+
+class AppModule(appModuleHandler.AppModule):
+
+	_shouldAnnounceResult = False
+	# Name change says the same thing multiple times for some items.
+	_resultsCache = ""
```

Comment:
This variable name suggests that it stores a cache of multiple results, which it doesn't.

```suggestion
	_cachedResult = ""
```

Reply:
@josephsl I don't think you've addressed this comment.